### ISSUE TITLE
Destructure keys

### ISSUE THREAD

**Issue by snoe** [clojure-lsp/clojure-lsp#905]
**Is your feature request related to a problem? Please describe.**
Destructuring keys after a number of `get`s have already been used can be a chore.

**Describe the solution you'd like**
Following clj-refactor would be a decent start.
https://github.com/clojure-emacs/clj-refactor.el/wiki/cljr-destructure-keys

Ideally:
Could destructure namespaced keywords to `::keys` / `::alias/keys` style. 
Handle `:syms` and `:strs` as well.

**Describe alternatives you've considered**
This could cycle destructuring/non-destructuring.

**Additional context**
More than top level destructuring should probably not occur by default but be behind a setting if supported.
`[params (get req :params) param1 (get params :param1)]` `{{:keys [param1]} :params}`

**Comment by ericdallo** [comment]
Sounds like a good refactoring! Liked the name too.

> More than top level destructuring should probably not occur by default but be behind a setting if supported.
[params (get req :params) param1 (get params :param1)] {{:keys [param1]} :params}

Could you elaborate a little bit this? with a more concise example and why a setting would be good?

**Comment by snoe** [comment]
You could imagine someone wanting to do this, I believe it is generally frowned upon nest destructurings.

```clojure
(defn handler [req]
  (let [params (get req :params) 
         param1 (get params :param1)]
    (inc param1)))
=>
(defn handler [{{:keys [param1]} :params}]
  (inc param1))
```

**Comment by ericdallo** [comment]
I see, a setting to allow nested destructuring, yeah, I'm not sure which one would good to be the default, maybe the not nested one indeed, makes sense!

**Comment by ericdallo** [comment]
Isn't that done @mainej ?

**Comment by mainej** [comment]
@ericdallo, yes, I think this could be closed. A refactoring named `destructure-keys` was implemented in #1139.

A few features were discussed here that weren't implemented in #1139. We could make separate tickets if those things are still needed:

* `destructure-keys` doesn't destructure `(get foo :bar)`. However, because of #1175, this can be converted to `(:bar foo)`, which can then be destructured.
* `destructure-keys` doesn't encourage nested destructuring. That said, it can be coerced into doing it. It won't help you write `(let [{|bar :bar} foo] (:baz bar))`, but if you do, then calling `destructure-keys` will lead to `(let [{{:keys [baz]} :bar} foo] baz)`
